### PR TITLE
Add nav_link widget for accessible navigation with active state

### PR DESCRIPTION
**Pull request by madmax983**
## Summary

Introduces a new `nav_link` widget for rendering navigation anchors that automatically mark themselves as active based on the current request path. This provides a reusable, accessible way to build navigation menus with proper ARIA attributes.

## Key Changes

- **New `nav_link` widget** (`autumn/src/widgets.rs`):
  - `nav_link()` function for exact path matching (default behavior)
  - `nav_link_matched()` function supporting configurable match modes via `NavLinkMatch` enum
  - `NavLinkMatch::Exact` - active only when paths match exactly
  - `NavLinkMatch::Prefix` - active when current path equals href or starts with href followed by `/` segment boundary
  - Automatically applies `class="active"` and `aria-current="page"` when active
  - Comprehensive test coverage including edge cases (trailing slashes, false positives, empty/root hrefs)

- **New `CurrentPath` extractor** (`autumn/src/extract.rs`):
  - Infallible extractor that provides the current request URI path (without query string)
  - Enables view helpers to determine which navigation link should be active
  - Includes tests verifying path extraction and query string stripping

- **Updated admin plugin** (`autumn-admin-plugin/src/templates.rs`):
  - Refactored `admin_layout` sidebar navigation to use `nav_link` widget
  - Eliminates manual active state management in favor of path-based comparison
  - Moves `JOBS_NAV_SLUG` and `RUNTIME_CONFIG_NAV_SLUG` constants to registry for reuse

- **Reserved sentinel slugs** (`autumn-admin-plugin/src/registry.rs`):
  - Added `JOBS_NAV_SLUG` (`"__admin_jobs"`) and `RUNTIME_CONFIG_NAV_SLUG` (`"__admin_config"`) as reserved model slugs
  - Prevents user models from hijacking built-in nav item active states
  - Includes validation tests ensuring these slugs cannot be registered

- **Updated prelude** (`autumn/src/prelude.rs`):
  - Exports `CurrentPath` extractor and `NavLinkMatch` enum for convenient access

## Implementation Details

- Prefix matching normalizes trailing slashes on `href` for consistent behavior (`/posts/` behaves identically to `/posts`)
- Empty or root-only hrefs never match in Prefix mode to prevent wildcard-like behavior
- Uses existing `crate::router::path_matches_route_prefix()` utility for segment-boundary-aware prefix matching
- All attributes are conditionally rendered using Maud's `class=[...]` and `aria-current=[...]` syntax to avoid emitting them when inactive

https://claude.ai/code/session_01TZc9ee4toV9xXv3CusqT1K

### CODE REVIEW

File: autumn/src/widgets.rs [hunk 1]
```diff
@@ -1108,6 +1109,108 @@ pub fn breadcrumb(crumbs: &[Crumb<'_>]) -> maud::Markup {
...
+        NavLinkMatch::Exact => current_path == href,
+        NavLinkMatch::Prefix => {
+            if current_path == href {
+                return true;
+            }
+            // Normalize a trailing slash so "/posts/" behaves the same as
+            // "/posts", but preserve a bare "/" rather than trimming it to an
+            // empty prefix — same convention as tenancy::is_public_path, since
+            // an empty prefix would otherwise match every absolute path.
+            let href = if href.len() > 1 {
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/src/widgets.rs:1204]: Not applying this one: `href` here is compared only against the single-byte, single-char literal `"/"`, and `trim_end_matches('/')` only ever strips that same single-byte ASCII slash. Since a codepoint can't have byte-length 1 without also having char-count 1 (and vice versa for this specific check), `.len() > 1` and `.chars().count() > 1` are equivalent for every possible input here — there's no href for which switching to `.chars().count()` changes the match outcome. `.len()` also matches the existing convention this code mirrors (`tenancy::is_public_path`), and avoids an O(n) UTF-8 walk on a per-nav-item render path for no behavioral gain. Happy to revisit if a concrete counter-example turns up.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: autumn/src/extract.rs [hunk 2]
```diff
@@ -736,6 +736,51 @@ mod tests {
...
+where
+    S: Send + Sync,
+{
+    type Rejection = std::convert::Infallible;
+
+    async fn from_request_parts(
+        parts: &mut axum::http::request::Parts,
+        _state: &S,
+    ) -> Result<Self, Self::Rejection> {
+        Ok(Self(parts.uri.path().to_owned()))
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/src/extract.rs]: Good catch, confirmed against axum's own `OriginalUri` docs — `Router::nest` strips the mount prefix from `parts.uri` inside the nested service, and `autumn-admin-plugin` itself mounts via `.nest(&prefix, router)`, so this would have silently broken `nav_link` for any nested handler. Fixed in 7604111: `CurrentPath` now reads `OriginalUri` (which falls back to `parts.uri` for non-nested requests), plus a regression test simulating a nested router.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: autumn-admin-plugin/src/templates.rs [hunk 3]
```diff
@@ -414,6 +412,19 @@ pub fn admin_layout(
...
+    // decides which sidebar item is active) and as the anchor's href below —
+    // so the "/jobs" and "/config" route suffixes each appear as a literal
+    // exactly once.
+    let jobs_href = format!("{prefix}/jobs");
+    let config_href = format!("{prefix}/config");
+    let current_path = match active_slug {
+        None => prefix.to_owned(),
+        Some(JOBS_NAV_SLUG) => jobs_href.clone(),
+        Some(RUNTIME_CONFIG_NAV_SLUG) => config_href.clone(),
+        Some(slug) => format!("{prefix}/{slug}"),
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn-admin-plugin/src/templates.rs:426]: Fair point on strict ARIA semantics, but leaving this as-is deliberately. `active_slug`/`current_path` here is a section-level construct by design (pre-existing behavior: the sidebar item was already highlighted via `class="active"` for every page under a model — list, detail, edit, history, import, export — before this PR; `nav_link` just adds `aria-current="page"` on top of that same signal). Making `aria-current` exact-page-accurate would mean threading the real per-page path through ~8 model sub-page handlers, which the source issue (#1134) explicitly scoped out ("active_slug threading removal is out of scope" — this PR's job was adding the primitive and proving it against the existing admin sidebar, not redesigning how admin_layout tracks state). Section-level `aria-current="page"` is also a common, defensible pattern in production admin panels. Happy to revisit as a follow-up if it's worth threading real paths through, but treating that as a separate, deliberate change rather than folding it into this PR.

---
_Generated by [Claude Code](https://claude.ai/code)_
